### PR TITLE
Update to forwarder 

### PR DESCRIPTION
**Pull request by joseph-fitz**
and adding support for ubuntu 18.04

Adding support for Circle 2.0, and as such the integration of workflows into the agent.

Branch builds will continue to work as they do currently.
For master branch builds, the packages as normal, and then wait until manual approval is given to upload the package via the circleci website. Once approval is granted the new packages will be uploaded to packagecloud.io

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+version: 2.0
+jobs:
+
+  build_rpm_packages:
```
Comment (emauton) [inline .circleci/config.yml]: Odd naming, given that this builds all packages? :thinking:

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    steps:
+      - restore_cache:
+          keys:
+            - hg_agent_package-{{ .Environment.CIRCLE_WORKFLOW_ID }}
+      - checkout
+      - run: gem install package_cloud
+      - run: make package-upload
+
+
+workflows:
```
Comment (emauton) [inline .circleci/config.yml]: This is a neat workflow description, easy to read.

File: Makefile [hunk 3]
```diff
@@ -84,12 +84,13 @@ package_test:
 	make -C targets/ubuntu-12.04
 	make -C targets/ubuntu-14.04
 	make -C targets/ubuntu-16.04
+	make -C targets/ubuntu-18.04
 
 deb-upload:
-	package_cloud push hostedgraphite/$(NAME)/$(DISTRO) out/deb/$(INIT)/$(NAME)_$(VERSION)_$(ARCH).deb
+	package_cloud push hostedgraphite/$(NAME)/$(DISTRO) /tmp/artifacts/out/deb/$(INIT)/$(NAME)_$(VERSION)_$(ARCH).deb
```
Comment (emauton) [inline Makefile]: Not a big fan of seeing `/tmp` here, but I can see what you're doing with the rest of the change.

Comment (joseph-fitz) [reply Makefile]: Yep, this was to try keep the change reasonably consistent with all of our other projects that use circle 2.0. I can switch the folder though, not a massive change at all!

Comment (emauton) [reply Makefile]: Nah, no need. Thanks!

**Pull request merged**